base/Enums.jl

```diff
@@ -140,4 +141,150 @@ macro enum(T,syms...)
     return blk
 end
 
+
+
+abstract FlagEnum <: AbstractEnum
+
+Base.convert{T<:Integer}(::Type{T}, x::FlagEnum) = convert(T, convert(Unsigned, x))
+for op in (:|, :&, :xor)
+    @eval function Base.$op{T<:FlagEnum}(x::T,y::T)
+        reinterpret(T, ($op)(convert(Unsigned, x), convert(Unsigned, y)))
+    end
+end
+Base.in{T<:FlagEnum}(x::T, y::T) = x & y == x
+Base.~{T<:FlagEnum}(x::T) = reinterpret(T, convert(Unsigned, typemax(T)) & ~convert(Unsigned, x))
+
+
+"""
+    @flagenum EnumName[::U] enumvalue1[=x] enumvalue1[=y]
+
+Create an [`FlagEnum`](:obj:`FlagEnum`) type with name `EnumName` and base member values
```

Comment:
now should use `@ref` syntax for cross-references, but there's no docstring for `FlagEnum`